File: app/src/main/java/jerry/filebrowser/util/NativeUtil.java
package jerry.filebrowser.util;

import jerry.filebrowser.file.FileAttribute;
import jerry.filebrowser.file.UnixFile;

public class NativeUtil {
    public static native boolean CreateFile(String path);

    public static native boolean CreateDir(String path);

    public static native boolean IsFileExist(String path);

    public static native UnixFile[] ListFiles(String path, int option);

    public static native boolean RenameFile(String from, String to);

    public static native boolean DeleteFile(String path);

    public static native boolean DeleteEmptyDir(String path);

    public static native int GetFileType(String path);

    public static native FileAttribute GetFileAttribute(String path);

    public static native int GetDisplay();
}

File: app/src/main/java/jerry/filebrowser/util/NativeUtil.kt
package jerry.filebrowser.util

import jerry.filebrowser.file.FileAttribute
import jerry.filebrowser.file.UnixFile

object NativeUtil {
    @JvmStatic
    external fun CreateFile(path: String?): Boolean

    @JvmStatic
    external fun CreateDir(path: String?): Boolean

    @JvmStatic
    external fun IsFileExist(path: String?): Boolean

    @JvmStatic
    external fun ListFiles(path: String?, option: Int): Array<UnixFile?>?

    @JvmStatic
    external fun RenameFile(from: String?, to: String?): Boolean

    @JvmStatic
    external fun DeleteFile(path: String?): Boolean

    // public static native boolean DeleteEmptyDir(String path);
    @JvmStatic
    external fun GetFileType(path: String?): Int

    @JvmStatic
    external fun GetFileAttribute(path: String?): FileAttribute?

    @JvmStatic
    external fun GetDisplay(): Int
}